mlir/python/PyCompilerDriver.cpp

```diff
@@ -78,6 +79,10 @@ PYBIND11_MODULE(compiler_driver, m)
         [](const char *source, const char *workspace, const char *moduleName, bool keepIntermediate,
            bool verbose, py::list pipelines,
            bool lower_to_llvm) -> std::unique_ptr<CompilerOutput> {
+            // Install signal handler to catch user interrupts (e.g. CTRL-C).
+            signal(SIGINT,
+                   [](int code) { throw std::runtime_error("KeyboardInterrupt (SIGINT)"); });
+
```

Comment:
One question I do have though is whether you need to change back the interrupt handler after exiting this function.

Reply:
I was wondering the same but when I tested it interrupts still worked fine in Python. I tested:
- running compiled func -> interrupt -> run some Python code -> interrupt again
- running compiled func to completion -> run some Python code -> interrupt

Reply:
You can try printing, flushing, and sleeping a couple of seconds just to see if it is the same interrupt or not.

Reply:
I'll resolve this conversation and leave it up to you to test this. Thanks David!

Reply:
> You can try printing, flushing, and sleeping a couple of seconds just to see if it is the same interrupt or not.

Ah do you think on the second interrupt from Python it could be calling this one still even the compiled program finished executing? Let me try :)

Reply:
Can confirm the interrupt handler triggered from Python after both running a compiled function to completion and interrupting a compiled function is always the Python handler, not the one I installed at the C++ layer.